t/api/04-jobs.t

```diff
@@ -689,6 +689,155 @@ subtest 'TEST is only mandatory parameter' => sub {
+        VERSION => '15-SP1',
+        FLAVOR  => 'Installer-DVD',
+        ARCH    => 'x86_64',
+        TEST    => 'autoupgrade',
+        MACHINE => '64bit',
+        BUILD   => '1234',
+        _GROUP  => 'opensuse',
+    );
+
+    $t->post_ok('/api/v1/jobs', form => \%new_jobs_post_params)->status_is(200);
+    my $result = $jobs->find($t->tx->res->json->{id})->settings_hash;
+    delete $result->{NAME};
+    is_deeply(
+        $result,
+        {
+            'QEMUCPU'             => 'qemu64',
+            'VERSION'             => '15-SP1',
+            'DISTRI'              => 'sle',
+            'MACHINE'             => '64bit',
+            'FLAVOR'              => 'Installer-DVD',
+            'ARCH'                => 'x86_64',
+            'BUILD'               => '1234',
+            'ISO_MAXSIZE'         => '4700372992',
+            'INSTALLONLY'         => 1,
+            'WORKER_CLASS'        => 'qemu_x86_64',
+            'DESKTOP'             => 'gnome',
+            'ISO'                 => 'SLE-15-SP1-Installer-DVD-64bit-Build1234-Media1.iso',
+            'BUILD_HA'            => '1234',
+            'TEST'                => 'autoupgrade',
+            'BETA'                => 1,
+            'BUILD_SES'           => '1234',
+            'BUILD_SDK'           => '1234',
+            'SHUTDOWN_NEEDS_AUTH' => 1,
+            'PATCH'               => 1,
+            'UPGRADE'             => 1,
+            'PUBLISH_HDD_1'       => 'SLES-15-SP1-x86_64-1234@64bit-minimal_with_sdk1234_installed.qcow2',
+            'ANOTHER_JOB'         => 'SLES-15-SP1-x86_64-1234@64bit-minimal_with_sdk1234_installed.qcow2',
+            'HDD_1'               => 'SLES-15-SP1-x86_64-1234@64bit-minimal_with_sdk1234_installed.qcow2',
+        },
+        'Job post method expand specified MACHINE, PRODUCT, TESTSUITE variable',
+    );
+};
+
+subtest 'circular reference settings' => sub {
+    my $products = $t->app->schema->resultset('Products');
+    $products->create(
+        {
+            version     => '12-SP5',
+            name        => '',
+            distri      => 'sle',
+            arch        => 'x86_64',
+            description => '',
+            flavor      => 'Installer-DVD',
+            settings    => [
+                {key => 'BUILD_SDK',           value => '%BUILD_HA%'},
+                {key => 'BETA',                value => '1'},
+                {key => 'ISO_MAXSIZE',         value => '4700372992'},
+                {key => 'BUILD_HA',            value => '%BUILD%'},
+                {key => 'BUILD_SES',           value => '%BUILD%'},
+                {key => 'SHUTDOWN_NEEDS_AUTH', value => '1'},
+                {
+                    key   => 'PUBLISH_HDD_1',
+                    value => 'SLES-%VERSION%-%ARCH%-%BUILD%@%MACHINE%-minimal_with_sdk%BUILD_SDK%_installed.qcow2'
+                },
+            ],
+        });
+    my $testsuites = $t->app->schema->resultset('TestSuites');
+    $testsuites->create(
```

Comment:
Just use `$t->app->schema->resultset('TestSuites')->create(`. Or - since there is another occurrence - move `my $testsutes = ...` to the top and use it at both places.